runtime/vm/resolvesupport.cpp

```diff
@@ -939,7 +939,15 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 					}
 					goto done;
 				}
-				if (!finalFieldSetAllowed(vmStruct, false, method, definingClass, classFromCP, field, canRunJavaCode)) {
+			if (
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				/* The withfield bytecode is allowed to set a final field. However, the invoker of withfield must have private access
+				 * to the field (similar to a constructor setting a final field) */
+				J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_FIELD_WITHER) &&
```

Comment:
This code doesn't seem to match the comment - all withfield of final fields will be allowed.

Reply:
The private access check is done above. So at this point all that needs to be checked is if it is a with_field.

I will update the comment to make that more clear